include/rxmesh/kernels/attribute.cuh

```diff
@@ -94,6 +94,79 @@ __launch_bounds__(blockSize) __global__
     }
 }
 
+struct CustomMaxPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value > a.value) ? b : a;
+    }
+};
+
+struct CustomMinPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value < a.value) ? b : a;
+    }
+};
+template <class T, uint32_t blockSize, typename HandleT, typename Operation>
+__launch_bounds__(blockSize) __global__
+    void arg_max_kernel(const Attribute<T, HandleT> X,
+                    uint32_t                    attribute_id,
+                    Operation                   op, //can be either max or min operation
+                    const uint32_t              num_patches,
+                    const uint32_t              num_attributes,
+                    cub::KeyValuePair<HandleT, T>*  d_block_output,
+                    uint32_t                    attribute_id)
+{
+    using LocalT = typename HandleT::LocalT;
+
+    assert(X.get_num_attributes() == 1); //we can only take arg max for a scalar attribute
+
+    uint32_t p_id = blockIdx.x;
+    if (p_id < num_patches) {
+        const uint16_t element_per_patch = X.size(p_id);
+        cub::KeyValuePair<HandleT, T> thread_val;
+        thread_val.value = std::numeric_limits<T>::lowest();
```

Comment:
This would work with `arg_max` but not with `arg_min`—you need `std::numeric_limits<T>::max()` in this case. I suggest to add a constexpr variable in the `CustomMaxPair`/`CustomMinPair` which how such a value. Then, the initial value would be simply be `thread_val.value = Operation::default_val`.